### ISSUE TITLE
Does Carveau support for Human vs Human 

### ISSUE THREAD

**Issue by takaaptech** [siggame/Cerveau#3]
Hi,
So far, i can setup two AI in a game. And i wonder if Carveau could support Human and Human or Human and AI in game? If yes, how can we implement that? Thank so much!

**Comment by JacobFischer** [comment]
It supports humans players in that the clients (Joueurs) that connect are really just sending and receiving data. Cerveau doesn't care if an AI or human is consuming the data, just that there is something at the other end.

Currently Cerveau is _heavily_ geared towards AI vs AI gameplay. It's a replacement for siggame's [old framework](https://github.com/siggame/MegaMinerAI-15), which is almost exclusively AI vs AI. With that in mind, human players are basically second class citizens in the design, we have many more hooks for AIs (like how much time they have to do calculations). The only real change for human players would be to increase the time clients have to make calculations over the entire game from 10 seconds to something like 30 minutes.

I have no examples of a human player in a client (Joueur), it's all just _theoretically_ possible. The visualizer that plays back Checkers games included with Cerveau is not our real Visualizer. We have other VIsualizers in private repos that play back our games. In addition, they hook into the client of their language so that can also playback a game as Cerveau sends it data, and takes input from humans to make commands to send to Cerveau. You could think of that like an AI that decides on what to send based on real-time human interaction.

Eventually we'll probably add a more formal example of a human client. Our old framework supports Chess, and that game has hacks in the old framework to allow for human players. If Cadre does succeed our old framework Cadre will need a chess human client, which we'll probably do by hooking up the JavaScript client to our private WebVisualizer.

tl;dr: It only supports humans in theory at the moment.

**Comment by takaaptech** [comment]
I got it, thank so much for detail explain!
